kasa/cli/discover.py

```diff
@@ -239,13 +239,29 @@ def _conditional_echo(label, value):
     _conditional_echo("Decrypted", pf(dr.decrypted_data) if dr.decrypted_data else None)
 
 
-async def find_host_from_alias(alias, target="255.255.255.255", timeout=1, attempts=3):
+async def find_dev_from_alias(
+    alias, credentials, target="255.255.255.255", timeout=1, attempts=3
+):
     """Discover a device identified by its alias."""
+    seen_ips = set()
     for _attempt in range(1, attempts):
-        found_devs = await Discover.discover(target=target, timeout=timeout)
-        for _ip, dev in found_devs.items():
+        found_devs = await Discover.discover(
+            target=target, timeout=timeout, credentials=credentials
+        )
+        for ip, dev in found_devs.items():
+            if ip in seen_ips:
+                continue
+            seen_ips.add(ip)
+            try:
+                await dev.update()
```

Comment:
@sdb9696 I don't have much experience with neither async python nor this library.

Should this update be given to `discover` with the `on_discovered` parameter?

In my mind, that would run the update immediately when a device is discovered, speeding up the process, while `await`ing the discovery in line 248 would only start the updates after the discovery is done.

Reply:
That's definitely a good optimization although it didn't previously do that. An `asyncio.event` and an `asyncio.task` is needed to be able to exit from the function immediately. The latest commit now does that.